### PR TITLE
fixed parsing of headers containing colons

### PR DESCRIPTION
**Pull request by czardoz**
Header parsing now happens only at the first occurance of `:`

### CODE REVIEW

**Comment by vegetableman** [comment]
@czardoz  It fails for the case `Mon, 25 Jul 2016 13:11:41 GMT`.

**Comment by vegetableman** [comment]
Header is passed a single value `Mon, 25 Jul 2016 13:11:41 GMT` instead of a key, value pair on instantiation.

File: lib/collection/header.js [hunk 1]
```diff
@@ -44,12 +44,7 @@ _.inherit((
      */
     Header = function PostmanHeader (options, name) {
         if (_.isString(options)) {
-            options = Header.parseSingle(options);
-        }
-
-        if (options && _.isString(name)) {
-            options.value = options.key;
-            options.key = name;
+            options = _.isString(name) ? { key: name, value: options } : Header.parseSingle(options);
```
Comment (shamasis) [inline lib/collection/header.js]: why this change?

File: lib/collection/header.js [hunk 2]
```diff
@@ -141,11 +136,15 @@ _.extend(Header, /** @lends Header */ {
      * @returns {{key: string, value: string}}
      */
     parseSingle: function (header) {
-        var parsed;
+        var parsed,
+            key,
+            value;
         parsed = header.split(':');
```
Comment (shamasis) [inline lib/collection/header.js]: why not regex split?

**Comment by shamasis** [comment]
Mostly LGTM.

File: lib/collection/header.js (hunk 1 above)
Comment (vegetableman) [inline lib/collection/header.js]: while running the tests, the parameter `options` is passed as `Mon, 25 Jul 2016 13:11:41 GMT`  and `name` as `Date`. `parseSingle` breaks for date values like `Mon, 25 Jul 2016 13:11:41 GMT` returning the value as instead `Mon, 25 Jul 2016 13`

File: lib/collection/header.js (hunk 2 above)
Comment (czardoz) [inline lib/collection/header.js]: Mostly because I find this simpler :-)

Comment (shamasis) [inline lib/collection/header.js]: Then use indexOf and substr if string. They're low level and faster.

File: test/functional/header.test.js [hunk 3]
```diff
@@ -50,6 +50,34 @@ describe('Header', function () {
...
+            var raw = 'name1: value1\r\nname2: value2',
+                list = new PropertyList(Header, {}, raw);
+            expect(list.all()).to.eql([{
+                key: 'name1',
+                value: 'value1'
+            }, {
+                key: 'name2',
+                value: 'value2'
+            }]);
+            expect(Header.unparse(list)).to.eql('name1: value1\nname2: value2')
```
Comment (elssar) [inline test/functional/header.test.js]: `to.be` instead of `to.eql`
